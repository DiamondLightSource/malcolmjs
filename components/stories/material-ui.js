import React from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import darkBaseTheme from 'material-ui/styles/baseThemes/darkBaseTheme';
import lightBaseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import AppBar from 'material-ui/AppBar';
import RaisedButton from 'material-ui/RaisedButton';
import FlatButton from 'material-ui/RaisedButton';
import Toggle from 'material-ui/Toggle';
import { storiesOf, action } from '@kadira/storybook';

function handleEdgeDeleteButton(){

  console.log("material-ui button clicked");

}

storiesOf('Button', module)
    .add('material-ui flat button', () => (
        <MuiThemeProvider  muiTheme={getMuiTheme(lightBaseTheme)}>

            <FlatButton key="edgeDeleteButton" className="pull-right btn btn-default" label="edge Delete Button"
                          onClick={handleEdgeDeleteButton()}>Delete Edge</FlatButton>
        </MuiThemeProvider>
  ))
    .add('material-ui raised button', () => (
        <MuiThemeProvider  muiTheme={getMuiTheme(darkBaseTheme)}>
            <RaisedButton key="edgeDeleteButton" className="pull-right btn btn-default" label="edgeDeleteButton"
                          onClick={action("Raised button clicked")}>Delete Edge</RaisedButton>
        </MuiThemeProvider>
    ))
    .add('material-ui Toggle switch', () => (
        <MuiThemeProvider  muiTheme={getMuiTheme(darkBaseTheme)}>
            <Toggle key="edgeToggleSwitch" className="pull-right btn btn-default" label="Edge Toggle"
                          onToggle={action("Toggle switch")}/>
        </MuiThemeProvider>
    ));

