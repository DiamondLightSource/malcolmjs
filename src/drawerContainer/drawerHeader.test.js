import React from 'react';
import { createShallow, createMount } from 'material-ui/test-utils';
import DrawerHeader from './drawerHeader.component';

describe('DrawerHeader', () => {
  let shallow;
  let mount;

  beforeEach(() => {
    shallow = createShallow();
    mount = createMount();
  });

  afterEach(() => {
    mount.cleanUp();
  });

  it('renders correctly', () => {
    const wrapper = shallow(
      <DrawerHeader
        closeAction={() => {}}
        popOutAction={() => {}}
        title="TTLIN1"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('closeAction calls the close method', () => {
    const closeAction = jest.fn();
    const wrapper = mount(
      <DrawerHeader
        title="TTLIN1"
        closeAction={closeAction}
        popOutAction={() => {}}
      />
    );

    wrapper
      .find('button')
      .first()
      .simulate('click');

    expect(closeAction.mock.calls.length).toEqual(1);
  });

  it('popOutAction calls the popOut method', () => {
    const popOutAction = jest.fn();
    const wrapper = mount(
      <DrawerHeader
        title="TTLIN1"
        popOutAction={popOutAction}
        closeAction={() => {}}
      />
    );

    wrapper
      .find('button')
      .last()
      .simulate('click');

    expect(popOutAction.mock.calls.length).toEqual(1);
  });
});
