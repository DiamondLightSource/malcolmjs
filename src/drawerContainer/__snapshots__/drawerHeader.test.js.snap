// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerHeader renders correctly 1`] = `
<div>
  <WithStyles(AppBar)
    position="static"
  >
    <WithStyles(Toolbar)
      style={
        Object {
          "display": "flex",
          "minWidth": 320,
          "padding": "0 8px",
        }
      }
    >
      <WithStyles(IconButton)
        data-cy="closedrawer"
        onClick={[Function]}
      >
        <pure(Close) />
      </WithStyles(IconButton)>
      <WithStyles(Typography)
        color="inherit"
        style={
          Object {
            "flexGrow": 1,
          }
        }
        variant="subheading"
      >
        TTLIN1
      </WithStyles(Typography)>
      <WithStyles(IconButton)
        onClick={[Function]}
      >
        <pure(OpenInNew) />
      </WithStyles(IconButton)>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
</div>
`;
